### PR TITLE
feat: Add Router-Authorization check

### PR DESCRIPTION
**Pull request by dbanty**
Aligns with the [subgraph security docs](https://github.com/apollographql/docs/pull/357) I've drafted

### CODE REVIEW

**Review by michael-watson — COMMENTED** [review]
I like this pattern because it enables you to change the secret in your deployments environment variables

**Comment by dbanty** [comment]
Yeah, the goal is for it to be completely deployment-agnostic—every secrets management technique I've seen supports injecting via env vars. I could see a world where folks want to deploy their subgraphs behind a load balancer / API gateway and do the enforcement at that level, but we can't help them much in the template in those cases 😅

File: src/index.ts [hunk 1]
```diff
@@ -6,18 +6,31 @@ import {
...
-  auth: req.headers.authorization,
-});
+> = async ({ req }) => {
+  if (routerSecret && req.headers["router-authorization"] !== routerSecret) {
+    throw new GraphQLError("Missing router authentication", {
+      extensions: {
+        code: "UNAUTHENTICATED",
+        http: { status: 401 },
+      },
+    });
```
Comment (patrick91) [inline src/index.ts:27]: wouldn't be best to use a plugin for this?

Comment (dbanty) [reply src/index.ts:27]: I don't know, I don't really use Apollo Server 😅. I based this off of [these docs](https://www.apollographql.com/docs/apollo-server/security/authentication#api-wide-authorization), if there's a better way then we should use it! (and update the docs)

**Comment by dbanty** [comment]
@patrick91 or @michael-watson, is there some alternative method we should use here (like a plugin), or can we merge this and copy over to the JavaScript template?

**Pull request merged**